File: m_storage_v2/mMVVM/loopwiki_java/src/main/java/com/example/gottgried/loopwiki_java/database/NoteDatabase.java
package com.example.gottgried.loopwiki_java.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.example.gottgried.loopwiki_java.database.daos.NoteDao;
import com.example.gottgried.loopwiki_java.database.models.Note;


// Room database class
@Database(entities = Note.class, version = 1, exportSchema = false)
public abstract class NoteDatabase extends RoomDatabase {
    //define static instance
    private static NoteDatabase mInstance;

    //method to get room database
    public static NoteDatabase getDatabase(Context context) {

        if (mInstance == null)
            mInstance = Room.databaseBuilder(context.getApplicationContext(),
                    NoteDatabase.class, "notes_db")
                    .build();

        return mInstance;
    }

    //method to remove instance
    public static void closeDatabase() {
        mInstance = null;
    }

    //define note dao ( data access object )
    public abstract NoteDao noteDao();


}

File: m_storage_v2/mMVVM/loopwiki/src/main/java/com/example/gottgried/loopwiki/database/NoteDatabase.kt
package com.example.gottgried.loopwiki.database

import android.arch.persistence.db.SupportSQLiteOpenHelper
import android.arch.persistence.room.Database
import android.arch.persistence.room.DatabaseConfiguration
import android.arch.persistence.room.InvalidationTracker
import android.arch.persistence.room.RoomDatabase
import com.example.gottgried.loopwiki.database.models.Note
import com.example.gottgried.loopwiki.database.daos.NoteDao
import android.arch.persistence.room.Room
import android.content.Context


// Room database class
@Database(entities = arrayOf(Note::class), version = 1, exportSchema = false)
abstract class NoteDatabase(var context: Context) : RoomDatabase() {

    //define static instance
    private var mInstance: NoteDatabase? = null


    init {
        if (mInstance == null)
            mInstance = Room.databaseBuilder(context.applicationContext,
                    NoteDatabase::class.java, "notes_db")
                    .build()
    }



    //method to remove instance
    fun closeDatabase() {
        mInstance = null
    }

    //define note dao ( data access object )
    abstract fun noteDao(): NoteDao



    override fun createOpenHelper(config: DatabaseConfiguration?): SupportSQLiteOpenHelper {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun createInvalidationTracker(): InvalidationTracker {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}